acme/provider.go

```diff
@@ -0,0 +1,7 @@
+package acme
+
+// ChallengeProvider makes the result of a challenge available to be solved
+type ChallengeProvider interface {
+	Present(domain, token, keyAuth string) error
+	CleanUp(domain string) error
```

Comment:
Yes, the second one would be the better signature imo. I feel like it shouldn't be required for providers to cache any of the information the Present hook provides.

@jehiah I'm sorry that it's taking me that long to review this, but I'm getting there, I promise :)